src/api/api_algebraic.cpp

```diff
@@ -418,4 +418,47 @@ extern "C" {
         Z3_CATCH_RETURN(0);
     }
 
+    unsigned Z3_API Z3_algebraic_get_poly_degree(Z3_context c, Z3_ast a) {
+        Z3_TRY;
+        LOG_Z3_algebraic_get_poly_degree(c, a);
+        RESET_ERROR_CODE();
+        CHECK_IS_ALGEBRAIC(a, 0);
+        algebraic_numbers::manager & _am = am(c);
+        algebraic_numbers::anum const & av = get_irrational(c, a);
+        return _am.degree(av);
+        Z3_CATCH_RETURN(0);
+    }
+
+    int64_t Z3_API Z3_algebraic_get_poly_coeff_int64(Z3_context c, Z3_ast a, unsigned i) {
```

Comment:
what about returning an expression of type Real? 
We can create a numeric constant that contains the value, and then other parts of the API can deal with extracting either int64_t, rationals, or infinite precision rationals.
We currently don't expose a stand-alone big-num object

Reply:
I changed it to  `Z3_ast_vector Z3_API Z3_algebraic_get_poly(Z3_context c, Z3_ast a)` returning a vector of coefficients.